### PR TITLE
Support for WLP-Activation-Type feature header for parallel activation

### PR DESCRIPTION
**Pull request by sebratton**

### CODE REVIEW

**Comment by sebratton** [comment]
#build

**Comment by sebratton** [comment]
@NottyCode Alasdair, Tom wanted me to double check that you are good with the more public facing aspect of his changes (new header name). Tom has a different PR he's currently working to use the WLP-Activation-Type in a subset of user feature.

File: dev/com.ibm.ws.kernel.boot.nested/src/com/ibm/ws/kernel/launch/internal/FrameworkConfigurator.java [hunk 1]
```diff
@@ -112,9 +112,10 @@ protected static void customize(BootstrapConfig config) {
         // It is not clear that multi-threading really helps with performance of the resolver.
         config.putIfAbsent("equinox.resolver.thread.count", "1");
 
-        // By default use single thread for activating bundles from start-level
-        // Set to -1 in bootstrap.properties results in using the number of threads == to num processors
-        config.putIfAbsent("equinox.start.level.thread.count", "1");
+        // By default use multiple threads for activating bundles from start-level
+        // Set to 1 in bootstrap.properties results sequentially starting bundles with a single thread
+        config.putIfAbsent("equinox.start.level.thread.count", "0");
```
[the remark this answers is not in the document]
Comment (sebratton) [reply dev/com.ibm.ws.kernel.boot.nested/src/com/ibm/ws/kernel/launch/internal/FrameworkConfigurator.java]: Update comment to describe meaning of '0' ==  Runtime.getRuntime().availableProcessors()

**Comment by sebratton** [comment]
#build

File: dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/Provisioner.java [hunk 2]
```diff
@@ -399,6 +401,9 @@ private Bundle installFeatureBundle(String urlString, String productName, Bundle
                     Region productRegion = getProductRegion(productName);
                     // Bundle will just be returned if something from this location exists already.
                     bundle = productRegion.installBundleAtLocation(location, new URL(urlString).openStream());
+                    if (ActivationType.parallel.equals(fr.getActivationType())) {
```
Comment (NottyCode) [inline dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/Provisioner.java]: ActivationType.parallel would appear from here to be a static final field, so why isn't it upper case as per the normal Java convention?

File: dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/provisioning/ActivationType.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.kernel.feature.provisioning;
+
+/**
+ *
+ */
+public enum ActivationType {
+    parallel,
```
Comment (NottyCode) [inline dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/provisioning/ActivationType.java]: I'd expect these values to be all upper case.

File: dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/Provisioner.java (hunk 2 above)
Comment (tjwatson) [reply dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/Provisioner.java]: Changed to uppercase constants

File: dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/provisioning/ActivationType.java (hunk 3 above)
Comment (tjwatson) [reply dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/provisioning/ActivationType.java]: Changed to upper case here.

**Comment by tjwatson** [comment]
#build

**Comment by tjwatson** [comment]
#build

**Pull request merged**